src/pages/PeoplePage.tsx

```diff
@@ -0,0 +1,75 @@
+import { useEffect, useMemo, useState } from 'react';
+import { useSearchParams } from 'react-router-dom';
+import { PeopleFilters } from '../components/PeopleFilters';
+import { Loader } from '../components/Loader';
+import { PeopleTable } from '../components/PeopleTable';
+import { Person } from '../types';
+import { getPeople } from '../api';
+import { getPreperedPeople } from '../utils/getPreperedPeople';
+import { getFilteredPeople } from '../utils/getFilteredPeople';
+
+export const PeoplePage = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+  const [searchParams] = useSearchParams();
+
+  useEffect(() => {
+    setIsLoading(true);
+
+    getPeople()
+      .then(loadedPeople => setPeople(getPreperedPeople(loadedPeople)))
+      .catch(error => setErrorMessage(error.message))
+      .finally(() => setIsLoading(false));
+  }, []);
+
+  const query = searchParams.get('query')?.toLowerCase() ?? '';
+  const selectedSex = searchParams.get('sex');
+  const selectedCenturies = useMemo(() => {
+    return searchParams.getAll('centuries');
+  }, [searchParams]);
+
+  const visiblePeople = useMemo(() => {
+    return getFilteredPeople(people, query, selectedCenturies, selectedSex);
+  }, [people, query, selectedCenturies, selectedSex]);
```

Comment:
Also, it's better to create a custom hook that returns the result of useMemo, than create a function and call it inside useMemo hooks.

```
export const useVisiblePeople = (params) => {
  return useMemo(() => {
   // filter logic
}, [bla])
}